### PR TITLE
[cleanup] Move artifact related methods to a dedicated module

### PR DESCRIPTION
**Pull request by Djailla**
## What changes are proposed in this pull request?

The projects/__init__.py file is pretty big, some parts can be extracted and pushed in a dedicated module. This is what I did for artifact related code.

## How is this patch tested?

Unit tests

### Is this a user-facing change?

- [x] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

### What component(s) does this PR affect?

- [ ] UI
- [ ] CLI
- [ ] API
- [ ] REST-API
- [ ] Examples
- [ ] Docs
- [ ] Tracking
- [ ] Projects
- [ ] Artifacts
- [ ] Models
- [ ] Scoring
- [ ] Serving
- [ ] R
- [ ] Java
- [x] Python

### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [ ] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

File: mlflow/projects/__init__.py [hunk 1]
```diff
@@ -841,80 +837,6 @@ def _get_docker_image_uri(repository_uri, work_dir):
...
-_artifact_storages = {
-    LocalArtifactRepository: _get_local_artifact_cmd_and_envs,
-    S3ArtifactRepository: _get_s3_artifact_cmd_and_envs,
-    AzureBlobArtifactRepository: _get_azure_blob_artifact_cmd_and_envs,
-    HdfsArtifactRepository: _get_hdfs_artifact_cmd_and_envs,
-    GCSArtifactRepository: _get_gcs_artifact_cmd_and_envs,
-}
-
-
 def _get_docker_artifact_storage_cmd_and_envs(artifact_uri):
```
Comment (sueann) [inline mlflow/projects/__init__.py]: would it make sense to also move this method to the artifact module? this is the method used by projects and `_artifact_storages` is only used by this method. then we might want to move the tests for `_get_docker_artifact_storage_cmd_and_envs` from `test_docker_projects` to something like `test_docker_artfacts` that match the new file better.

**Comment by Djailla** [comment]
@sueann Hello, any update on this one ?

**Comment by sueann** [comment]
hi @Djailla, not sure if you saw, I had made one comment above^ what do you think?

**Comment by Djailla** [comment]
@sueann I did the change you proposed ;)

**Comment by sueann** [comment]
ah sorry i don't see the change - maybe missed a commit?

**Pull request closed without merging**